src/lib/crypto/DY.Lib.Crypto.SplitPredicate.fst

```diff
@@ -50,12 +50,12 @@ let split_crypto_predicate_parameters_to_split_function_parameters (params:split
     Some (tag, (tr, key, data))
   );
 
-  local_fun_t = params.local_pred_t;
+  local_fun_t = mk_dependent_type params.local_pred_t;
   global_fun_t = params.global_pred_t;
 
   default_global_fun = params.mk_global_pred always_false;
 
-  apply_local_fun = params.apply_local_pred;
+  apply_local_fun = (fun lpred x -> params.apply_local_pred lpred x);
```

Comment:
1) (Why) Do we need this lambda expression?
2) If we do, can we find a more telling name for `x`?

Reply:
Because there is an implicit hidden there: it is actually
```fstar
  apply_local_fun = (fun #tag_set lpred x -> params.apply_local_pred lpred x);
```
so that if we want to eta-reduce it, it would look like
```fstar
  apply_local_fun = (fun #tag_set -> params.apply_local_pred);
```

We could also have
```fstar
  apply_local_fun = (fun lpred trace_key_and_data -> params.apply_local_pred lpred trace_key_and_data);
```
or
```fstar
  apply_local_fun = (fun lpred (tr, key, data) -> params.apply_local_pred lpred (tr, key, data));
```
whichever you prefer of the options above.

Reply:
Done!